guides/theming.md

```diff
@@ -1,311 +1,359 @@
 
-Additionally, in Angular Material, a configuration may optionally include `typography` settings.
-More information on how typography works can be [found in a dedicated guide][3].
+### Sass
 
-Angular Material theme styles are generated _statically_ at build-time so that your
-app doesn't have to spend cycles generating theme styles when bootstrapping.
+Angular Material's theming APIs are built with [Sass](https://sass-lang.com). This document assumes
+familiary with CSS and Sass basics, including variables, functions, and mixins.
 
-## Using a pre-built theme
+You can use Angular Material without Sass by using a pre-built theme, described in
+[Using a pre-built theme](#using-a-pre-built-theme) below. However, using the library's Sass API
+directly gives you the most control over the styles in your application.
 
-Angular Material comes prepackaged with several pre-built theme css files. These theme files also
-include all the styles for the core (styles common to all components), so you only have to include a
-single css file for Angular Material in your app.
+## Palettes
 
-You can include a theme file directly into your application from
-`@angular/material/prebuilt-themes`
+A **palette** is a collection of colors representing a portion of color space. Each value in this
+collection is called a **hue**. In Material Design, the hues in a palette are numbered from zero
+to 900, lightest to darkest.
 
-Available pre-built themes:
-* `deeppurple-amber.css`
-* `indigo-pink.css`
-* `pink-bluegrey.css`
-* `purple-green.css`
+Angular Material represents a palette as a [Sass map][sass-maps]. This map contains the
+palette's hues and another nested map of contrast colors for each of the hues. The contrast colors
+serve as text color when using a hue as a background color. The example below demonstrates the
+structure of a palette.
 
-If you're using Angular CLI, this is as simple as including one line
-in your `styles.css`  file:
-```css
-@import '@angular/material/prebuilt-themes/deeppurple-amber.css';
+```scss
+$indigo: (
+ 50: #e8eaf6,
+ 100: #c5cae9,
+ 200: #9fa8da,
+ 300: #7986cb,
+ // ... continues to 900
+ contrast: (
+   50: rgba(black, 0.87),
+   100: rgba(black, 0.87),
+   200: rgba(black, 0.87),
+   300: white,
+   // ... continues to 900
+ )
+);
 ```
 
-Alternatively, you can just reference the file directly. This would look something like:
-```html
-<link href="node_modules/@angular/material/prebuilt-themes/indigo-pink.css" rel="stylesheet">
+[sass-maps]: https://sass-lang.com/documentation/values/maps
+
+### Predefined palettes
+
+Angular Material offers 19 predefined palettes based on the 2014 version of the Material Design
+spec. See the [Material Design 2014 color palettes][2014-palettes] for a full list.
+
+In addition to hues numbered from zero to 900, the 2014 Material Design palettes each include
+distinct _accent_ hues numbered as `A100`, `A200`, `A400`, and `A700`.
```

Comment:
Do we need to explain what the difference is between accent and normal hues?

Reply:
I added a sentence calling out the fact that they're not strictly required but can be optionally used when defining a theme.